xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -215,13 +228,31 @@ public void updateLocalityStore(Map<XdsLocality, LocalityInfo> localityInfoMap)
 
       updatePicker(newState, childPickers);
 
+      // There is a race between picking a subchannel and updating localities, which leads to
+      // the possibility that RPCs will be sent to a removed locality. As a result, those RPC
+      // loads will not be recorded. We consider this to be natural. By removing locality counters
+      // after updating subchannel pickers, we further narrow down the window in which this race
```

Comment:
By scheduling in syncContext you actually eliminate the race.

Reply:
You are right. Changed the comment.